File: app/src/main/java/com/raisesail/mvp/z_mvp_final/base/BasePresenter.java
package com.raisesail.mvp.z_mvp_final.base;

import com.raisesail.mvp.z_mvp_final.IBasePresenter;

import java.lang.ref.WeakReference;

public class BasePresenter<V> implements IBasePresenter<V> {

    protected WeakReference<V> mWeakReference;
    /**
     * 页面构建
     * @param view
     */
    @Override
    public void registerView(V view) {
        mWeakReference = new WeakReference<>(view);
    }

    /**
     * 页面销毁
     */
    @Override
    public void unRegisterView() {
        if (mWeakReference != null){
            mWeakReference.clear();
            mWeakReference = null;
        }
    }
    /**
     * 获取页面的引用
     */
    @Override
    public V getView(){
        return mWeakReference != null ? mWeakReference.get() : null;
    }

    /**
     * 判断页面是否销毁，接口返回设置数据必须调用此方法判断界面是否还存在
     */
    @Override
    public boolean isViewAttached(){
        return mWeakReference != null && mWeakReference.get()!= null;
    }

}

File: app/src/main/java/com/raisesail/mvp/z_mvp_final_kotlin/base/BasePresenter.kt
package com.raisesail.mvp.z_mvp_final_kotlin.base

import com.raisesail.mvp.z_mvp_final_kotlin.IPresenter
import com.raisesail.mvp.z_mvp_final_kotlin.IView
import java.lang.ref.WeakReference

open class BasePresenter<V : IView> : IPresenter<V> {

    protected var mWeakReference : WeakReference<V> ?= null

    override fun isAttachView(): Boolean {
        return false
    }

    override fun unRegisterView() {
        mWeakReference?.clear()
    }

    override fun registerView(view: V) {
        mWeakReference = WeakReference<V>(view)
    }

    fun getView():V?{
        return mWeakReference?.get()
    }
}